### PR TITLE
TW-2042 Update contacts screen ui

### PR DESCRIPTION
**Pull request by KhaledNjim**
## Ticket
 #2042 
## need merge
[LInagora design flutter pr](https://github.com/linagora/linagora-design-flutter/pull/50)

## Resolved
**Attach screenshots or videos demonstrating the changes**
- Web:
Screenshot_1

- Android:
2042_mobile

- IOS:

### CODE REVIEW

**Comment by hoangdat** [comment]
- [ ] still not see the screenshot of contacts which load from address book @KhaledNjim

File: lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart [hunk 1]
```diff
@@ -26,103 +29,126 @@ class ExpansionContactListTile extends StatelessWidget {
...
-                child: Avatar(
-                  mxContent: snapshot.data?.avatarUrl,
-                  name: contact.displayName,
-                  size: 48.0,
+    return Container(
+      decoration: BoxDecoration(
+        border: Border(
+          bottom: BorderSide(
+            color: LinagoraStateLayer(LinagoraSysColors.material().surfaceTint)
+                .opacityLayer3,
```
Comment (hoangdat) [inline lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart:37]: IMO, it is `surface` color ( ref from `chat list item`)
image

Please create a util class to make us reuse it.

File: lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart [hunk 2]
```diff
@@ -26,103 +29,126 @@ class ExpansionContactListTile extends StatelessWidget {
...
+              children: [
+                IgnorePointer(
+                  child: Avatar(
+                    mxContent: snapshot.data?.avatarUrl,
+                    name: contact.displayName,
+                  ),
+                ),
+                const SizedBox(
+                  width: 12.0,
                 ),
```
Comment (hoangdat) [inline lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart:61]: it should be 8

image

File: lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart [hunk 3]
```diff
@@ -26,103 +29,126 @@ class ExpansionContactListTile extends StatelessWidget {
...
+                              child: Row(
+                                children: [
+                                  Flexible(
+                                    child: BuildDisplayName(
+                                      profileDisplayName:
+                                          snapshot.data?.displayName,
+                                      contactDisplayName: contact.displayName,
+                                      highlightKeyword: highlightKeyword,
+                                      style: LinagoraTextStyle.material()
+                                          .bodyMedium2
```
Comment (hoangdat) [inline lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart:80]: you use it exactly, but when I run in web app, the weight of this text is not precise

image

Comment (hoangdat) [reply lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart:80]: and color is not true too.

File: lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart [hunk 4]
```diff
@@ -26,103 +29,126 @@ class ExpansionContactListTile extends StatelessWidget {
...
+                        ),
+                      if (contact.email != null)
+                        HighlightText(
+                          text: contact.email!,
+                          searchWord: highlightKeyword,
+                          style: Theme.of(context)
+                              .textTheme
+                              .bodyMedium
+                              ?.copyWith(
+                                color: LinagoraRefColors.material().neutral[60],
```
Comment (hoangdat) [inline lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart:131]: wrong color 
image

File: lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart [hunk 5]
```diff
@@ -26,103 +29,126 @@ class ExpansionContactListTile extends StatelessWidget {
...
+                        ),
+                      if (contact.phoneNumber != null)
+                        HighlightText(
+                          text: contact.phoneNumber!,
+                          searchWord: highlightKeyword,
+                          style: Theme.of(context)
+                              .textTheme
+                              .bodyMedium
+                              ?.copyWith(
+                                color: LinagoraRefColors.material().neutral[60],
```
Comment (hoangdat) [inline lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart:142]: idem?

File: lib/pages/contacts_tab/contacts_appbar_style.dart [hunk 6]
```diff
@@ -1,24 +1,40 @@
...
+          ? Alignment.center
+          : AlignmentDirectional.centerStart;
 
   static EdgeInsetsDirectional appbarPadding =
       const EdgeInsetsDirectional.symmetric(
     horizontal: 16.0,
   );
 
   static EdgeInsetsDirectional contentPadding = EdgeInsetsDirectional.zero;
+  static const double textStyleHeight = 24 / 17;
```
Comment (hoangdat) [inline lib/pages/contacts_tab/contacts_appbar_style.dart:29]: what is it? can you explain it?

**Comment by nqhhdev** [comment]
Wrong hint text

Screenshot 2024-10-01 at 12 48 52

**Comment by nqhhdev** [comment]
Missing padding

Screenshot 2024-10-01 at 12 51 26

**Comment by KhaledNjim** [comment]
> * [ ]  still not see the screenshot of contacts which load from address book @KhaledNjim
  
The contacts at the bottom are from my phonebook

[2042_phoneBook.webm](https://github.com/user-attachments/assets/fd38b033-946a-4749-bf40-e15b6c4ab340)

File: lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart (hunk 1 above)
Comment (KhaledNjim) [reply lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart:37]: This is the color for the seperator between items, its the same as in chat list 
 contact list 
image
chat list 
image

File: lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart (hunk 2 above)
Comment (KhaledNjim) [reply lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart:61]: This sizedBox is between the name and chips such owner or not-activated, its not displayed in the design, should i change it to 8 ?
image

**Comment by nqhhdev** [comment]
- Need to move hover style, divider style, selected background to system design

File: lib/pages/contacts_tab/contacts_appbar_style.dart (hunk 6 above)
Comment (KhaledNjim) [reply lib/pages/contacts_tab/contacts_appbar_style.dart:29]: The default height of the text( 17 is this case) will be multiplied by this, to make the height of the text 24
image

File: lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart (hunk 2 above)
Comment (KhaledNjim) [reply lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart]: fixed

File: lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart (hunk 3 above)
Comment (KhaledNjim) [reply lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart]: fixed

File: lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart (hunk 4 above)
Comment (KhaledNjim) [reply lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart]: fixed

File: lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart (hunk 5 above)
Comment (KhaledNjim) [reply lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart]: fixed

**Comment by hoangdat** [comment]
- please apply also system-design to this PR @KhaledNjim

**Comment by hoangdat** [comment]
- please keep align to this in top. we should not keep it in center because if we have only 2 lines, blink blink will occur

File: lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart [hunk 7]
```diff
@@ -26,103 +28,124 @@ class ExpansionContactListTile extends StatelessWidget {
...
-            crossAxisAlignment: CrossAxisAlignment.start,
-            children: [
-              IgnorePointer(
-                child: Avatar(
-                  mxContent: snapshot.data?.avatarUrl,
-                  name: contact.displayName,
-                  size: 48.0,
+    return TwakeListItem(
+      child: Padding(
+        padding: const EdgeInsets.only(left: 8.0, top: 8.0, bottom: 8.0),
```
Comment (dab246) [inline lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart:33]:
```suggestion
        padding: const EdgeInsetsDirectional.only(start: 8.0, top: 8.0, bottom: 8.0),
```

File: lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart [hunk 8]
```diff
@@ -26,103 +28,124 @@ class ExpansionContactListTile extends StatelessWidget {
...
+        child: FutureBuilder<Profile?>(
+          key: contact.matrixId != null ? Key(contact.matrixId!) : null,
+          future: contact.status == ContactStatus.active
+              ? getProfile(context)
+              : null,
+          builder: (context, snapshot) {
+            return Row(
+              crossAxisAlignment: CrossAxisAlignment.center,
```
Comment (dab246) [inline lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart:41]: `CrossAxisAlignment.center` is the default value so we don't need to set it again.

File: lib/pages/search/search_text_field.dart [hunk 9]
```diff
@@ -23,7 +23,7 @@ class SearchTextField extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return Material(
-      borderRadius: BorderRadius.circular(16.0),
+      borderRadius: BorderRadius.circular(24.0),
```
Comment (dab246) [inline lib/pages/search/search_text_field.dart:26]:
```suggestion
      borderRadius: const BorderRadius.all(Radius.circular(24.0)),
```

File: lib/widgets/sliver_expandable_list.dart [hunk 10]
```diff
@@ -19,26 +17,12 @@ class SliverExpandableList extends StatefulWidget {
...
-              );
-            }
-            return widget.itemBuilder(context, index - 1);
-          },
-        );
+    return SliverList.builder(
+      itemCount: widget.itemCount,
+      itemBuilder: (context, index) {
+        return widget.itemBuilder(context, index);
       },
```
Comment (dab246) [inline lib/widgets/sliver_expandable_list.dart:26]:
```suggestion
      itemBuilder: widget.itemBuilder,
```

**Comment by nqhhdev** [comment]
I found issue related text overflowed on iPhone 15

Simulator Screenshot - iPhone 15 - 2024-10-07 at 15 30 30

**Comment by nqhhdev** [comment]
Height only 79px, pls check it

Screenshot 2024-10-07 at 15 32 53

**Comment by nqhhdev** [comment]
Missing hover

Screenshot 2024-10-07 at 15 39 57

**Comment by KhaledNjim** [comment]
> Height only 79px, pls check it
> 
> Screenshot 2024-10-07 at 15 32 53

please make sure to run pub get , because i fixed this in linagora-design-flutter

**Comment by nqhhdev** [comment]
> > Height only 79px, pls check it
> > Screenshot 2024-10-07 at 15 32 53
> 
> please make sure to run pub get , because i fixed this in linagora-design-flutter

SURE!

**Comment by nqhhdev** [comment]
Missing hover on web

Screenshot 2024-10-07 at 15 45 17

**Comment by nqhhdev** [comment]
> > Missing hover on web: Contact and Chat list
> Hover is grey and selected item is blue


I mean missing border !

**Comment by nqhhdev** [comment]
Pls add PR of `linagora-design-flutter` lib on this PR

**Comment by nqhhdev** [comment]
How about this case?

- For example: my account has 1000 characters, what is happened?

Simulator Screenshot - iPhone 15 - 2024-10-08 at 10 28 03

**Comment by hoangdat** [comment]
Hi @KhaledNjim , please focus on the alignment, bottom text is well align with bottom avatar.

Screenshot 2024-10-08 at 18 15 00

**Comment by KhaledNjim** [comment]
> Hi @KhaledNjim , please focus on the alignment, bottom text is well align with bottom avatar.
> 
> Screenshot 2024-10-08 at 18 15 00

Fixed
image

**Review by nqhhdev — APPROVED** [review]
I'm tested on Web and Mobile, LGTM

File: assets/images/ic_person_check.svg
Comment (hoangdat) [inline assets/images/ic_person_check.svg:1]: why not use it from `Icons`?

File: lib/pages/contacts_tab/contacts_tab_body_view.dart [hunk 11]
```diff
@@ -347,6 +349,7 @@ class _SliverPhonebookLoading extends StatelessWidget {
   }
 }
 
+// ignore: unused_element
```
Comment (hoangdat) [inline lib/pages/contacts_tab/contacts_tab_body_view.dart:352]: why need it? why not delete the widget?

File: lib/pages/contacts_tab/contacts_tab_view_style.dart [hunk 12]
```diff
@@ -7,5 +7,10 @@ class ContactsTabViewStyle {
   static const double loadingSpacer = 16;
   static const double avatarSize = 48.0;
 
-  static const searchItemsHoverRadius = BorderRadius.all(Radius.circular(12));
+  static const searchItemsHoverRadius = BorderRadius.all(Radius.circular(4));
+  static const invitefriendContainterPadding = EdgeInsets.all(8);
+  static const invitefriendContainterLeadingIconPadding = EdgeInsets.all(8);
+  static const leadingIconDimension = 24.0;
+  static const invitefriendTextPadding =
```
Comment (hoangdat) [inline lib/pages/contacts_tab/contacts_tab_view_style.dart:14]:
```suggestion
  static const inviteFriendContainterPadding = EdgeInsets.all(8);
  static const inviteFriendContainterLeadingIconPadding = EdgeInsets.all(8);
  static const leadingIconDimension = 24.0;
  static const inviteFriendTextPadding =
```

File: lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart [hunk 13]
```diff
@@ -26,103 +28,132 @@ class ExpansionContactListTile extends StatelessWidget {
...
+                                        contactDisplayName: contact.displayName,
+                                        highlightKeyword: highlightKeyword,
+                                        style: LinagoraTextStyle.material()
+                                            .bodyMedium2
+                                            .copyWith(
+                                              color: Theme.of(context)
+                                                  .colorScheme
+                                                  .onSurface,
+                                              fontWeight: FontWeight.w600,
+                                            ),
```
Comment (hoangdat) [inline lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart:82]: why not move it to System design ?

Comment (hoangdat) [reply lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart:82]: please move it to the System design and use it in: chat list item, contact list item,

File: lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart [hunk 14]
```diff
@@ -26,103 +28,132 @@ class ExpansionContactListTile extends StatelessWidget {
...
+                        HighlightText(
+                          text: contact.matrixId!,
+                          searchWord: highlightKeyword,
+                          style: Theme.of(context)
+                              .textTheme
+                              .bodyMedium
+                              ?.copyWith(
+                                color:
+                                    LinagoraRefColors.material().tertiary[30],
+                              ),
```
Comment (hoangdat) [inline lib/pages/new_private_chat/widget/expansion_contact_list_tile.dart:120]: move it to System design too

File: lib/pages/contacts_tab/contacts_appbar_style.dart [hunk 15]
```diff
@@ -1,24 +1,40 @@
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/utils/responsive/responsive_utils.dart';
 import 'package:flutter/cupertino.dart';
 
 class ContactsAppbarStyle {
+  static ResponsiveUtils responsiveUtils = getIt.get<ResponsiveUtils>();
```
Comment (hoangdat) [inline lib/pages/contacts_tab/contacts_appbar_style.dart:6]: not a good idea, please inject it in method. Fx: `static AlignmentGeometry alignmentTitle(responseUtils, context)`

File: lib/pages/contacts_tab/contacts_appbar_style.dart [hunk 16]
```diff
@@ -1,24 +1,40 @@
...
 
   static const double toolbarHeight = 56.0;
 
   static const double leadingWidth = 76.0;
 
   static const double textFieldBorderRadius = 24.0;
 
-  static AlignmentGeometry alignment = AlignmentDirectional.centerStart;
+  static AlignmentGeometry alignmentTitle(context) =>
+      ResponsiveUtils().isMobile(context)
```
Comment (hoangdat) [inline lib/pages/contacts_tab/contacts_appbar_style.dart:19]: Dont create instance here

File: lib/pages/contacts_tab/contacts_tab_body_view.dart [hunk 17]
```diff
@@ -14,16 +14,19 @@ import 'package:fluffychat/presentation/model/contact/get_presentation_contacts_
...
 import 'package:fluffychat/widgets/sliver_expandable_list.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:flutter_svg/flutter_svg.dart';
 import 'package:linagora_design_flutter/linagora_design_flutter.dart';
 import 'package:matrix/matrix.dart';
 
 class ContactsTabBodyView extends StatelessWidget {
   final ContactsTabController controller;
+  final bool enableFriendsInvite = false;
```
[the remark this answers is not in the document]
Comment (hoangdat) [reply lib/pages/contacts_tab/contacts_tab_body_view.dart:27]: did you forget?

File: lib/pages/contacts_tab/contacts_appbar_style.dart (hunk 16 above)
Comment (hoangdat) [reply lib/pages/contacts_tab/contacts_appbar_style.dart:19]: done

File: lib/pages/contacts_tab/contacts_appbar_style.dart (hunk 15 above)
Comment (hoangdat) [reply lib/pages/contacts_tab/contacts_appbar_style.dart:6]: applied as other style file

File: lib/pages/contacts_tab/contacts_appbar_style.dart [hunk 18]
```diff
@@ -1,24 +1,40 @@
...
   static EdgeInsetsDirectional appbarPadding =
       const EdgeInsetsDirectional.symmetric(
     horizontal: 16.0,
   );
 
   static EdgeInsetsDirectional contentPadding = EdgeInsetsDirectional.zero;
+  static const double textStyleHeight = 24 / 17;
+  static const EdgeInsets trailingIconPadding = EdgeInsets.only(right: 16.0);
+  static EdgeInsets titlePadding(context) => EdgeInsets.only(
+        left: ResponsiveUtils().isMobile(context) ? 0 : 16.0,
```
Comment (hoangdat) [inline lib/pages/contacts_tab/contacts_appbar_style.dart:32]: idem

File: lib/pages/contacts_tab/contacts_tab_body_view.dart [hunk 19]
```diff
@@ -347,64 +347,6 @@ class _SliverPhonebookLoading extends StatelessWidget {
   }
 }
 
-class _SliverRecentContacts extends StatelessWidget {
```
Comment (hoangdat) [inline lib/pages/contacts_tab/contacts_tab_body_view.dart:350]: why remove it?

File: lib/pages/contacts_tab/contacts_tab_view_style.dart [hunk 20]
```diff
@@ -7,5 +7,10 @@ class ContactsTabViewStyle {
   static const double loadingSpacer = 16;
   static const double avatarSize = 48.0;
 
-  static const searchItemsHoverRadius = BorderRadius.all(Radius.circular(12));
+  static const searchItemsHoverRadius = BorderRadius.all(Radius.circular(4));
```
Comment (hoangdat) [inline lib/pages/contacts_tab/contacts_tab_view_style.dart:10]: not found where use it

**Comment by KhaledNjim** [comment]
Rebased and tested for android and web, LGTM

**Comment by hoangdat** [comment]
- [ ] search result not show item in recent list 

https://github.com/user-attachments/assets/ddbfe99e-6fa0-4ccf-9717-bfedd3084c5d

File: lib/utils/extension/presentation_search_extension.dart [hunk 21]
```diff
@@ -0,0 +1,38 @@
...
+    return email?.toLowerCase().contains(keyword.toLowerCase()) ?? false;
+  }
+
+  bool _matchedContactInfo(String keyword) {
+    return _matchedName(keyword) ||
+        _matchedEmail(keyword) ||
+        _matchedMatrixId(keyword);
+  }
+
+  bool doesMatchKeyword(String keyword) {
```
[the remark this answers is not in the document]
Comment (nqhhdev) [reply lib/utils/extension/presentation_search_extension.dart:23]: Updated

File: lib/presentation/mixins/contacts_view_controller_mixin.dart [hunk 22]
```diff
@@ -402,49 +402,51 @@ mixin class ContactsViewControllerMixin {
...
             if (presentationRecentContactNotifier.isDisposed) return;
 
-            if (Matrix.of(context).twakeSupported) {
-              if (keyword == null) {
-                presentationRecentContactNotifier.value =
-                    recent.take(_defaultLimitRecentContacts).toList();
-                return;
-              }
-              presentationRecentContactNotifier.value = [];
+            if (keyword == null || keyword.isEmpty) {
```
Comment (hoangdat) [inline lib/presentation/mixins/contacts_view_controller_mixin.dart:405]: Test cases:
- not existed recent item in tom, expect `presentationRecentContactNotifier` have the same list response from `_searchRecentChatInteractor`
- recent list have one item duplicated in tom contact, expect `presentationRecentContactNotifier` have list but not include duplicated item

File: test/utils/presentation_search_extension_test.dart [hunk 23]
```diff
@@ -0,0 +1,126 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:fluffychat/presentation/model/search/presentation_search.dart';
+import 'package:fluffychat/utils/extension/presentation_search_extension.dart';
+
+void main() {
+  group('doesMatchKeyword test', () {
+    test(
+        'WHEN displayName is null'
+        'AND keyword is keyword'
```
Comment (hoangdat) [inline test/utils/presentation_search_extension_test.dart:9]: AND no keyword in neither mail nor matrixId

File: test/utils/presentation_search_extension_test.dart [hunk 24]
```diff
@@ -0,0 +1,126 @@
...
+        matrixId: 'matrix_id_1',
+        email: 'email1@gmail.com',
+        displayName: null,
+      );
+      expect(search.doesMatchKeyword('keyword'), false);
+    });
+
+    test(
+        'WHEN email is null'
+        'AND keyword is keyword'
```
Comment (hoangdat) [inline test/utils/presentation_search_extension_test.dart:21]: AND no keyword in neither displayName nor matrixId

File: test/utils/presentation_search_extension_test.dart [hunk 25]
```diff
@@ -0,0 +1,126 @@
...
+    test(
+        'WHEN keyword matches directChatMatrixID'
+        'THEN returns true', () {
+      const search = ContactPresentationSearch(
+        matrixId: 'matrix_id_1',
+        email: 'email1@gmail.com',
+        displayName: 'display_name_1',
+      );
+      expect(search.doesMatchKeyword('matrix_id'), false);
+    });
```
Comment (hoangdat) [inline test/utils/presentation_search_extension_test.dart:64]: - please add logic `doesMatchKeyword` for `matrixId` and `directMatrixId`

File: test/utils/presentation_search_extension_test.dart [hunk 26]
```diff
@@ -0,0 +1,126 @@
...
+        'WHEN email is null'
+        'AND keyword is keyword'
+        'THEN returns false', () {
+      const search = ContactPresentationSearch(
+        matrixId: 'matrix_id_1',
+        email: null,
+        displayName: 'display_name_1',
+      );
+      expect(search.doesMatchKeyword('keyword'), false);
+    });
```
Comment (hoangdat) [inline test/utils/presentation_search_extension_test.dart:88]: Duplicated with case 2

File: test/utils/presentation_search_extension_test.dart [hunk 27]
```diff
@@ -0,0 +1,126 @@
...
+        'WHEN keyword matches displayName'
+        'AND displayName is display_name_1'
+        'THEN returns true', () {
+      const search = ContactPresentationSearch(
+        matrixId: 'matrix_id_1',
+        email: 'email1@gmail.com',
+        displayName: 'display_name_1',
+      );
+      expect(search.doesMatchKeyword('display_name'), true);
+    });
```
Comment (hoangdat) [inline test/utils/presentation_search_extension_test.dart:100]: Duplicated case 3

File: test/utils/presentation_search_extension_test.dart [hunk 28]
```diff
@@ -0,0 +1,126 @@
...
+        'WHEN keyword matches email'
+        'AND email is email1@gmail.com'
+        'THEN returns true', () {
+      const search = ContactPresentationSearch(
+        matrixId: 'matrix_id_1',
+        email: 'email1@gmail.com',
+        displayName: 'display_name_1',
+      );
+      expect(search.doesMatchKeyword('email1'), true);
+    });
```
Comment (hoangdat) [inline test/utils/presentation_search_extension_test.dart:112]: Replace with case: all fields have `keyword`

File: test/utils/presentation_search_extension_test.dart [hunk 29]
```diff
@@ -0,0 +1,126 @@
...
+        'AND keyword is non_matching_keyword'
+        'THEN returns false', () {
+      const search = ContactPresentationSearch(
+        matrixId: 'matrix_id_1',
+        email: 'email1@gmail.com',
+        displayName: 'display_name_1',
+      );
+      expect(search.doesMatchKeyword('non_matching_keyword'), false);
+    });
+  });
```
Comment (hoangdat) [inline test/utils/presentation_search_extension_test.dart:125]: remove

File: test/utils/presentation_search_extension_test.dart [hunk 30]
```diff
@@ -17,110 +16,126 @@ void main() {
...
       );
       expect(search.doesMatchKeyword('keyword'), false);
     });
 
     test(
-        'WHEN keyword matches displayName'
-        'AND displayName is display_name_1'
-        'THEN returns true', () {
+        'WHEN displayName matches keyword'
+        'THEN returns false', () {
```
Comment (hoangdat) [inline test/utils/presentation_search_extension_test.dart:31]: return true

File: test/utils/presentation_search_extension_test.dart [hunk 31]
```diff
@@ -17,110 +16,126 @@ void main() {
...
+        'WHEN keyword is empty'
+        'THEN returns false', () {
       const search = ContactPresentationSearch(
         matrixId: 'matrix_id_1',
         email: 'email1@gmail.com',
-        displayName: 'display_name_1',
+        displayName: 'matrix_user',
       );
-      expect(search.doesMatchKeyword('email1'), true);
+      expect(search.doesMatchKeyword(''), true);
```
Comment (hoangdat) [inline test/utils/presentation_search_extension_test.dart:48]: false?

File: test/utils/presentation_search_extension_test.dart [hunk 32]
```diff
@@ -17,110 +16,126 @@ void main() {
...
     test(
-        'WHEN keyword does not match any contact info'
-        'AND keyword is non_matching_keyword'
+        'WHEN email matches keyword'
         'THEN returns false', () {
       const search = ContactPresentationSearch(
         matrixId: 'matrix_id_1',
         email: 'email1@gmail.com',
-        displayName: 'display_name_1',
+        displayName: 'matrix_user',
```
Comment (hoangdat) [inline test/utils/presentation_search_extension_test.dart:80]: IMO, it should be `email`, not displayName

File: test/utils/presentation_search_extension_test.dart [hunk 33]
```diff
@@ -17,110 +16,126 @@ void main() {
...
+        'WHEN keyword is empty'
         'THEN returns false', () {
       const search = ContactPresentationSearch(
         matrixId: 'matrix_id_1',
-        email: null,
-        displayName: 'display_name_1',
+        email: 'email1@gmail.com',
+        displayName: 'matrix_user',
+      );
+      expect(search.doesMatchKeyword(''), true);
```
Comment (hoangdat) [inline test/utils/presentation_search_extension_test.dart:93]: false?

File: test/utils/presentation_search_extension_test.dart [hunk 34]
```diff
@@ -17,110 +16,126 @@ void main() {
...
+    });
+  });
+
+  group('doesMatchKeyword test with matrixId', () {
```
Comment (hoangdat) [inline test/utils/presentation_search_extension_test.dart:97]: do we need test for the field `directChatMatrixID`?

File: test/utils/presentation_search_extension_test.dart [hunk 35]
```diff
@@ -138,4 +138,15 @@ void main() {
...
+
+  test(
+      'WHEN all fields matches keyword'
+      'THEN returns true', () {
+    const search = RecentChatPresentationSearch(
+      displayName: 'matrix',
+      directChatMatrixID: 'direct_chat_matrix_1',
+      roomId: 'matrix_1',
+    );
+    expect(search.doesMatchKeyword('matrix_'), false);
```
Comment (hoangdat) [inline test/utils/presentation_search_extension_test.dart:150]: why it is false?

**Review by hoangdat — APPROVED** [review]

**Pull request merged**